arangod/Aql/Optimizer/Rule/OptimizerRulePushFilterVectorIndex.cpp

```diff
@@ -0,0 +1,201 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2025 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Business Source License 1.1 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jure Bajic
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Aql/Ast.h"
+#include "Aql/AstHelper.h"
+#include "Aql/Condition.h"
+#include "Aql/ExecutionNode/EnumerateNearVectorNode.h"
+#include "Aql/ExecutionNode/EnumerateCollectionNode.h"
+#include "Aql/Expression.h"
+#include "Aql/ExecutionNode/CalculationNode.h"
+#include "Aql/ExecutionNode/MaterializeRocksDBNode.h"
+#include "Aql/ExecutionNode/LimitNode.h"
+#include "Aql/ExecutionNode/FilterNode.h"
+#include "Aql/ExecutionNode/SortNode.h"
+#include "Aql/Optimizer.h"
+#include "Aql/OptimizerRules.h"
+#include "Aql/OptimizerUtils.h"
+#include "Aql/QueryContext.h"
+#include "Basics/Exceptions.h"
+#include "Indexes/Index.h"
+
+using namespace arangodb;
+using namespace arangodb::aql;
+using EN = arangodb::aql::ExecutionNode;
+
+namespace {
+
+#define LOG_RULE_ENABLED false
+#define LOG_RULE_IF(cond) LOG_DEVEL_IF((LOG_RULE_ENABLED) && (cond))
+#define LOG_RULE LOG_RULE_IF(true)
+
+std::unique_ptr<Expression> removeFilterNode(
```

Comment:
Perhaps we can use a better name that indicates that this function only _tries_ to remove the filter node.